Haskell/Haskell.sublime-syntax

```diff
@@ -1,407 +1,1388 @@
+    - include: big-arrow-operators
+    - include: left-arrow-operators
+    - include: right-arrow-operators
+    - include: prefix-parens-operators
+    - include: infix-quoted-operators
+    # Match all not otherwise matched single quotes as promoition operator
+    # Note: Found in real world code but not in specs so far.
+    - match: \'
       scope: keyword.operator.haskell
-    - match: \b(\d+)(?:(\.)(\d+)([eE][-+]?\d+)?|([eE][-+]?\d+))\b
-      scope: meta.number.float.decimal.haskell
-      captures:
-        1: constant.numeric.value.haskell
-        2: punctuation.separator.decimal.haskell
-        3: constant.numeric.value.haskell
-        4: constant.numeric.value.exponent.haskell
-        5: constant.numeric.value.exponent.haskell
-    - match: '{{operator_infix}}'
+    - match: '{{operator_double_colon}}'
+      scope: keyword.operator.double-colon.haskell
+    - match: '{{operator_symbol}}'
       scope: keyword.operator.haskell
-    - include: operator_paren
 
-  operator_paren:
-    - match: \(({{operator_parens}})\)
-      scope: variable.function.infix.haskell
-      captures:
-        1: keyword.operator.haskell
+  big-arrow-operators:
+    - match: '{{operator_big_arrow}}'
+      scope: keyword.operator.big-arrow.haskell
 
-  preprocessor:
-    # In addition to Haskell's "native" syntax,
-    # GHC permits the C preprocessor to be run on a source file.
-    - match: ^\s*(#)\s*\w+
-      scope: meta.preprocessor.c
+  left-arrow-operators:
+    - match: '{{operator_left_arrow}}'
+      scope: keyword.operator.arrow.haskell
+
+  right-arrow-operators:
+    - match: '{{operator_right_arrow}}'
+      scope: keyword.operator.arrow.haskell
+
+  prefix-parens-operators:
+    - match: (\()\s*({{operator_parens}})\s*(\))
+      scope: meta.prefix.haskell
       captures:
-        1: punctuation.definition.preprocessor.c
-    - include: pragma
+        1: punctuation.definition.prefix.begin.haskell
+        2: keyword.operator.haskell
+        3: punctuation.definition.prefix.end.haskell
 
-  string:
-    - match: '"'
-      scope: punctuation.definition.string.begin.haskell
-      push:
-        - meta_include_prototype: false
-        - meta_scope: string.quoted.double.haskell
-        - match: $|"
-          scope: punctuation.definition.string.end.haskell
-          pop: true
-        - match: '{{escape_sequence}}'
-          captures:
-            1: constant.character.escape.haskell
-            2: constant.character.escape.decimal.haskell
-            3: constant.character.escape.octal.haskell
-            4: constant.character.escape.hexadecimal.haskell
-            5: constant.character.escape.control.haskell
-    - match: |-
-        (?x)
-        (')
-        (?:
-          [\ -\[\]-~]            # Basic Char
-          | {{escape_sequence}}  # Escapes
-        )
-        ([^']*)
-        (')
-      scope: string.quoted.single.haskell
+  infix-quoted-operators:
+    - match: (`)\s*([\w'.]*(#*))\s*(`)
```

Comment:
Already thought about it, yes, but honestly was too lazy to rework those at this point, because we'd need to push into context then in order to correctly scope all `.` as accessor etc.. Or is there only one level of namespace?

Reply:
> Or is there only one level of namespace?

Not confident enough to answer this, so most likely we would need a subcontext here with a look-head or branch, depending on what is easier.